// @prec-ui/resizable-handle/components/ResizableHandle.jsx
import React from 'npm:react@^18.3.1'
import { Stack, Divider } from 'npm:@mui/material@^6.2.1'
import { DragIndicatorOutlined } from 'npm:@mui/icons-material@^6.2.1'

const ResizableCustomHandle = ({hoverColer = '#3F77AF', topMargin = '0px', bottomMargin = '0px'}) => {

    return (
        <Stack
            sx={{
                height: '100%',
            }}
            alignItems={'center'}
        >
            <Divider
                orientation='vertical'
                variant='middle'
                textAlign='center'
                sx={{
                    marginTop: topMargin,
                    width: '0px',
                    zIndex: 1000,
                    '&:hover, &:active': {
                        "&::before, &::after": {
                            borderColor: hoverColer,
                            borderWidth: '5px'
                        },
                    },
                    display: 'flex',
                    alignItems: 'center',
                    justifyContent: 'center'

                }}
            >
                <Stack>
                    <DragIndicatorOutlined color='action' sx={{ width: '20px', color: hoverColer }} />
                </Stack>
            </Divider>
        </Stack>

    )
}

export default ResizableCustomHandle
