// @prec-ui/resizable-handle/components/ResizableHandle.js
import React from "react";
import { Stack, Divider } from "@mui/material";
import { DragIndicatorOutlined } from "@mui/icons-material";
const ResizableCustomHandle = ({ hoverColer = '#3F77AF', topMargin = '0px', bottomMargin = '0px' })=>{
  return /*#__PURE__*/ React.createElement(Stack, {
    sx: {
      height: '100%'
    },
    alignItems: 'center'
  }, /*#__PURE__*/ React.createElement(Divider, {
    orientation: "vertical",
    variant: "middle",
    textAlign: "center",
    sx: {
      marginTop: topMargin,
      width: '0px',
      zIndex: 1000,
      '&:hover, &:active': {
        "&::before, &::after": {
          borderColor: hoverColer,
          borderWidth: '5px'
        }
      },
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    }
  }, /*#__PURE__*/ React.createElement(Stack, null, /*#__PURE__*/ React.createElement(DragIndicatorOutlined, {
    color: "action",
    sx: {
      width: '20px',
      color: hoverColer
    }
  }))));
};
export default ResizableCustomHandle;
